### PR TITLE
4.4 Solo Challenge

### PR DESCRIPTION
**Pull request by elainefirefly**
Requesting code review @Bontgoy

### CODE REVIEW

File: ruby/vampires.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  age = gets.chomp.to_i
+  puts "What year were you born?"
+  birth_year = gets.chomp.to_i
+  puts "Our company cafeteria serves garlic bread. Should we order some for you? Type 'y' or 'n'."
+  order_garlic = gets.chomp == 'y' ? true : false;
+  puts "Would you like to enroll in the company's health insurance? Type 'y' or 'n'."
+  enroll_insurance = gets.chomp == 'y' ? true : false;
+
+  # Release 2
+  correct_age = Time.now.year - birth_year == age
```
Comment (Bontgoy) [inline ruby/vampires.rb]: Great work with `Time.now.year`!!!!!

🎉

File: ruby/vampires.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  result = "Definitely a vampire" if name == "Drake Cula" || name == "Tu Fang"
+  result ||= "Results inconclusive"
+
+  # Release 4
+  allergy = ""
+  until allergy == "done"
+    puts "Do you have allergies? Type each one and press enter. Type 'done', when list is complete. : "
+    allergy = gets.chomp
+    if allergy == "sunshine"
+      result = "Probably a vampire"
```
Comment (Bontgoy) [inline ruby/vampires.rb]: This looks good but I would organize your allergy checking to happen before your vampire results are decided.

An important way to consider designing a workflow is to consider which steps should be taken in which order.

In this case I think a good approach would be to gather all of the user information first and then do you vampire evaluations.

File: ruby/vampires.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  birth_year = gets.chomp.to_i
+  puts "Our company cafeteria serves garlic bread. Should we order some for you? Type 'y' or 'n'."
+  order_garlic = gets.chomp == 'y' ? true : false;
+  puts "Would you like to enroll in the company's health insurance? Type 'y' or 'n'."
+  enroll_insurance = gets.chomp == 'y' ? true : false;
+
+  # Release 2
+  correct_age = Time.now.year - birth_year == age
+  result = nil
+  if correct_age && order_garlic && enroll_insurance
```
Comment (Bontgoy) [inline ruby/vampires.rb]: Consider nesting your control flow.  In this case if the we know the employee's name is Drac of Fang then we don't need to run anymore of these conditions. 

```
if name == "Drake Cula" || name == "Tu Fang"
  results = "Definitely a vampire"
else
  # run other checks
end
```

**Comment by elainefirefly** [comment]
Hi Tyler,
I would have done those things, but I felt restricted by this constraint that was on the first release:
**Match the following conditions in the order they're listed.** Your program should base its result on the latest condition matched, not the first condition matched. In other words, it's not a "only one condition will apply" sort of scenario. This may mean repeatedly updating a variable as each condition is checked, so that the variable always contains the most precise result.
So even, though it made a lot of sense to check for the name first and at the same time combine the allergy question in release 4 before giving the evaluation, I chose to follow the constraint of the first release to match the order as they were listed on the module.

**Comment by Bontgoy** [comment]
@elainefirefly Thanks for checking in. I definitely understand and appreciate your feedback.

Ill deliver this info to curriculum to see if we can make changes for the next cohorts.  Don't worry though.  This was a great submission!

**Comment by elainefirefly** [comment]
Thanks Tyler!

**Pull request merged**